src/features/island/buildings/components/building/BuildingOilTank.tsx

```diff
@@ -0,0 +1,325 @@
+    if (!isCookingBuilding(buildingName)) {
+      return 0;
+    }
+
+    const dailyOilConsumption = BUILDING_DAILY_OIL_CONSUMPTION[buildingName];
+    const hoursPerDay = 24;
+    const oilTimeInHours = (oil / dailyOilConsumption) * hoursPerDay;
+
+    // Convert hours to milliseconds
+    const millisecondsPerHour = 60 * 60 * 1000;
+    const oilTimeInMillis = oilTimeInHours * millisecondsPerHour;
+
+    return oilTimeInMillis;
+  }
+
+  const handleSupplyOil = (amount: number) => {
+    gameService.send("cookingOil.supplied", {
+      building: buildingName,
+      buildingId,
+      oilQuantity: amount,
+    });
+
+    gameService.send("SAVE");
+
+    setShowAddOilModal(false);
+  };
+
+  const canAddOil = () => {
+    if (!isCookingBuilding(buildingName)) {
+      return false;
+    }
+
+    return (
+      oilRemainingInBuilding <= BUILDING_DAILY_OIL_CAPACITY[buildingName] - 1
+    );
+  };
+
+  const canIncrementOil = () => {
+    if (!canAddOil() || !isCookingBuilding(buildingName)) return false;
+
+    const oilBalance = game.inventory.Oil ?? new Decimal(0);
+    const hasEnoughOil = oilBalance.toNumber() >= totalOilToAdd;
+    const buildingNotFull =
+      oilRemainingInBuilding + totalOilToAdd <=
+      BUILDING_DAILY_OIL_CAPACITY[buildingName] - 1;
+
+    return hasEnoughOil && buildingNotFull;
+  };
+
+  const calculatePercentageFull = (buildingName: BuildingName) => {
+    const totalOilMillis = calculateOilTimeRemaining() * 1000;
+
+    if (!isCookingBuilding(buildingName)) {
+      return 0;
+    }
+
+    const buildingTimeCapacity = getOilTimeInMillis(
+      BUILDING_DAILY_OIL_CAPACITY[buildingName]
+    );
+
+    const bla = (totalOilMillis / buildingTimeCapacity) * 100;
+
+    return bla;
+  };
+
+  const calculateOilTimeRemaining = () => {
+    if (!isCookingBuilding(buildingName)) {
+      return 0;
+    }
+
+    const readyAt = building?.crafting?.readyAt;
+
+    if (!readyAt || !currentlyCooking)
+      return getOilTimeInMillis(oilRemainingInBuilding) / 1000;
+
+    const secondsTillReady = (readyAt - Date.now()) / 1000;
+
+    const cookingSeconds = COOKABLES[currentlyCooking]?.cookingSeconds;
+
+    const oilInRecipe = building?.crafting?.boost?.["Oil"] ?? 0;
+
+    const percentageCooked = secondsTillReady / cookingSeconds;
+
+    const oilAllocatedRemaining = oilInRecipe * percentageCooked;
+
+    // Calculate the building remaining oil time
+    const remainingOilTime =
+      getOilTimeInMillis(oilRemainingInBuilding + oilAllocatedRemaining) / 1000;
+
+    return remainingOilTime;
+  };
+
+  const oilInTank = calculatePercentageFull(buildingName);
+  const runtime = calculateOilTimeRemaining();
+  const boostPercentage =
+    BUILDING_OIL_BOOSTS[buildingName as CookingBuildingName] * 100;
+
+  useUiRefresher();
+
+  const onClick = () => {
+    setShowAddOilModal(true);
+  };
+  return (
+    <>
+      <div className="relative w-full">
+        {runtime === 0 ? (
+          <Label
```

Comment:
<img width="197" alt="Screenshot 2024-05-28 at 2 59 34 PM" src="https://github.com/sunflower-land/sunflower-land/assets/11745561/3323c9f3-f95f-48f7-8d71-e8b69546d6bf">

A few suggestions:

1. Use the default state. The red usually indicates an error
2. Include the secondary Icon so players know they can press on it
3. "No oil" -> "Oil boost"